src/mon/PGMap.h

```diff
@@ -283,10 +290,16 @@ class PGMap : public PGMapDigest {
   void update_pool_deltas(
     CephContext *cct,
     const utime_t ts,
-    const mempool::pgmap::unordered_map<uint64_t, pool_stat_t>& pg_pool_sum_old);
+    const mempool::pgmap::unordered_map<int64_t, pool_stat_t>& pg_pool_sum_old);
   void clear_delta();
 
   void deleted_pool(int64_t pool) {
+    for (auto i = pool_stat.begin();  i != pool_stat.end(); ++i) {
+      if (i->first.first == pool) {
+	pool_stat.erase(i);
+      }
```

Comment:
this, and all fo the make_pair stuff, could be avoided with nested maps: map<int64_t,map<int32_t,pool_stat_t>>

Reply:
That's how I originally implemented that. In fact it complicates insertion, dumping, encoding/decoding instead. Given the fact that we don't care much about the performance on pool/osd removal I'd prefer the current implementation.